### PR TITLE
Upgrade to JupyterLab by reinitializing project structure

### PR DESCRIPTION
**Pull request by philippjfr**
Another attempt at this, this time by reinitializing the project and then readding the old code.

### CODE REVIEW

**Comment by philippjfr** [comment]
Have to figure out the tests but locally this now works in classic notebook and jlab when following the dev instructions. Ping @mattpap and @bryevdv for testing.

**Comment by philippjfr** [comment]
Actually, the nbextension doesn't seem to work yet.

**Comment by philippjfr** [comment]
Fixed, should all work now.

**Comment by philippjfr** [comment]
Tests finally passing as well.

**Comment by bryevdv** [comment]
I have publish `3.0.0-dev1` to NPM and PyPI. Seems to moslty be working:

Screen Shot 2021-03-21 at 12 07 28 PM

A few notes:

* the "loading BokehJS" message does not complete in subsequent loads after a first (but things still function)
* `push_notebook` does not work (no comms available)
* Things seem to work with or without the extension installed?? I vaguely recall Jupyter relenting on JS exec so maybe the classic notebook codepaths just work in that case? Is there a way to verify that the actual extension version is being used?

**Comment by bryevdv** [comment]
Well, actually looking at the console, I think it is not using the extension at all, but must be falling back to the bare JS exec that classic uses?

Screen Shot 2021-03-21 at 12 17 38 PM

**Comment by bryevdv** [comment]
@jasongrout Do you have any ideas about the above? It seems I am seeing that error and others are not. But I think I have everything mentioned in other threads I can find installed:
```
(230_39) ❯ jupyter labextension list
JupyterLab v3.0.12
/Users/bryan/anaconda/envs/230_39/share/jupyter/labextensions
        @bokeh/jupyter_bokeh v3.0.0-dev1 enabled OK (python, jupyter_bokeh)
        @jupyter-widgets/jupyterlab-manager v4.0.0-alpha.1 enabled OK (python, jupyterlab_widgets)


~/work/bokeh/examples/howto bryanv/hasprops*
(230_39) ❯ conda list ipywidgets
# packages in environment at /Users/bryan/anaconda/envs/230_39:
#
# Name                    Version                   Build  Channel
ipywidgets                8.0.0a4                  pypi_0    pypi

~/work/bokeh/examples/howto bryanv/hasprops*
(230_39) ❯ pip install jupyterlab_widgets
Requirement already satisfied: jupyterlab_widgets in /Users/bryan/anaconda/envs/230_39/lib/python3.9/site-packages (2.0.0a1)
```
How can I debug this?

**Comment by bryevdv** [comment]
OK seems like it was the 2.0.0a1 version of  jupyterlab-widgets   that the `pip install --pre jupyter_bokeh` brought it that was busting everything for me

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@bokeh/jupyter_bokeh",
-  "version": "3.0.0-dev1",
+  "version": "3.0.0-dev2",
```
Comment (mattpap) [inline package.json:3]: It should be `3.0.0-dev.2` to keep it consistent with semver and other bokeh's npm packages.

Comment (bryevdv) [reply package.json:3]: FWIW the server regex validator that the semver project links to in its docs validates the above version, which is why i used it.

**Comment by jasongrout** [comment]
> OK seems like it was the 2.0.0a1 version of jupyterlab-widgets that the `pip install --pre jupyter_bokeh` brought it that was busting everything for me

Yes, that `--pre` switch bitten me too before. Glad you figured it out.

I took `jupyterlab_bokeh` for a quick test run and it seems to work. Thanks for updating it!

**Comment by philippjfr** [comment]
Thanks @jasongrout. One quick question, for a conda-forge recipe should we include jupyter-packaging as a dependency or will it skip invoking the build step as long as the package on PyPI has already bundled it?

**Comment by jasongrout** [comment]
> Thanks @jasongrout. One quick question, for a conda-forge recipe should we include jupyter-packaging as a `host` dependency or will it skip invoking the build step as long as the package on PyPI has already bundled it?

I think because setup.py imports it, it needs to be a build dependency, like here:

https://github.com/conda-forge/jupyterlab-fasta-feedstock/blob/5d32434a19e6ca5c7a8dd3705db8e1bf770de2c2/recipe/meta.yaml#L20

**Pull request merged**